pluginLauncher/lib/source/DobbyRdkPluginDependencySolver.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2021 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/*
+ * File:   DobbyRdkPluginDependencySolver.cpp
+ *
+ */
+#include "DobbyRdkPluginDependencySolver.h"
+
+#include <Logging.h>
+
+#include <boost/graph/topological_sort.hpp>
+
+// -----------------------------------------------------------------------------
+/**
+ * @brief Adds a plugin to the solver.
+ *
+ * Each plugin must be known to the solver - added by this method before its dependencies
+ * are tracked.
+ *
+ * @param[in]   name   Name of the plugin
+ *
+ * @return True if the plugin has been added successfully,
+ * false otherwise (the plugin had already been added).
+ *
+ */
+bool DobbyRdkPluginDependencySolver::addPlugin(const std::string &name)
+{
+    AI_LOG_FN_ENTRY();
+
+    if (mDescriptorMap.count(name) != 0)
+    {
+        AI_LOG_WARN("Plugin %s already added to the solver", name.c_str());
+        return false;
+    }
+
+    // Each plugin is represented by a vertex of a directed graph.
+    const VertexDescriptor descriptor = boost::add_vertex(mDependencyGraph);
+    // Store the name of the plugin as a property of the vertex.
+    boost::put(boost::vertex_name_t{}, mDependencyGraph, descriptor, name);
+    mDescriptorMap.insert(StringVertexDescriptorMap::value_type{ name, descriptor });
+
+    AI_LOG_FN_EXIT();
+    return true;
+}
+
+// -----------------------------------------------------------------------------
+/**
+ * @brief Adds a dependency between two plugins to the solver.
+ *
+ * If plugin A depends on plugin B, then the call is:
+ * @code addDependency(A, B); @endcode
+ *
+ * @param[in]   pluginName       The name of the plugin which depends on @p dependencyName.
+ * @param[in]   dependencyName   The name of the plugin on which @p pluginName depends.
+ *
+ * @return True if the dependency has been added successfully,
+ * false otherwise (one of the plugins had not been added to the solver).
+ *
+ */
+bool DobbyRdkPluginDependencySolver::addDependency(const std::string &pluginName, const std::string &dependencyName)
+{
+    AI_LOG_FN_ENTRY();
+
+    if (mDescriptorMap.count(pluginName) == 0)
```

Comment:
These checks are case sensitive - plugin names should be case insensitive. Convert `pluginName` to lowercase before checking

Reply:
Done.